scalding-core/src/test/scala/com/twitter/scalding/ReferencedClassFinderTest.scala

```diff
@@ -0,0 +1,49 @@
+package com.twitter.scalding
+
+import org.apache.hadoop.io.BytesWritable
+import org.scalatest.{ Matchers, WordSpec }
+
+case class C1(a: Int)
+case class C2(b: Int)
+case class C3(c: Int)
+case class C4(d: Int)
+
+trait TraitType {
```

Comment:
1 and 2 should be fine, at least based on my testing.

Primitives seem to work, but in `setSerialization` it filters out primitives and array types:
```
    val kryoClasses = withKryo.getKryoRegisteredClasses
      .filterNot(_.isPrimitive) // Cascading handles primitives and arrays
      .filterNot(_.isArray)```

I'll add those filters to `findReferencedClasses`